Add new catalog entry for ndx-nirs

dsleiter opened the pull request:
This is the `ndx-nirs` extension that we developed initially during the NWB+DANDI hackathon. Thanks for the review, and looking forward to feedback!

rly commented:
@dsleiter Thanks for adding `ndx-nirs` to the catalog! I went through the source code with a fine-toothed comb and it all looks great!

My one comment is to please test it with PyNWB 1.5.1, HDMF 2.5.6 and to allow a range of `pynwb>=1.5.1,<2; hdmf>=2.5.6,<3` so that it works with future minor and bugfix versions of PyNWB and HDMF. And if a change is made, please update the version number and release on PyPI.

When you are ready for this to be merged (it can be merged now as is if you want), let me know and I will merge it into the catalog.

dsleiter commented:
Thanks for the review @rly!

I updated to the newest versions of hdmf and pynwb and configured the version matching as you suggested, and then released the new version on pypi and updated the version number here. I think it is all ready to be merged into the catalog.

We look forward to continued updates to and support for the extension!
